airbyte-server/src/main/java/io/airbyte/server/ConfigDumpImporter.java

```diff
@@ -107,7 +105,17 @@ public ConfigDumpImport(String initialVersion,
     }
   }
 
-  public ImportRead importData(File archive) {
+  public ImportRead importData(String targetVersion, File archive) {
+    return importDataInternal(targetVersion, archive, Optional.empty());
+  }
+
+  public ImportRead importDataWithSeed(String targetVersion, File archive, Path seedPath) {
+    return importDataInternal(targetVersion, archive, Optional.of(seedPath));
+  }
+
+  // seedPath - if present, merge with the import. otherwise just use the data in the import.
+  private ImportRead importDataInternal(String targetVersion, File archive, Optional<Path> seedPath) {
```

Comment:
i know, it's an optional as arg. i contained it to a private method, and i think this gives the clearest indication of how this method is supposed to work.

Reply:
blasphemy

Reply:
jk 🚀